### PR TITLE
[Merged by Bors] - feat(Algebra/Module/LinearMap/Defs): add `restrictScalars` as a `LinearMap`

### PR DESCRIPTION
**Pull request by judithludwig**
add some missing lemmas on `RestrictScalars` as a `Linear Map`.

This contribution was created as part of the AIM workshop "Formalizing algebraic geometry" in
June 2024.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/LinearMap/Defs.lean [hunk 1]
```diff
@@ -1038,4 +1038,41 @@ end Module
...
+
+@[simp]
+theorem restrictScalars_smul (c : R₁) (f : M →ₗ[S] N) :
+    (c • f).restrictScalars R = c • f.restrictScalars R :=
+  rfl
+
+variable (S M N R R₁)
+
+/-- `LinearMap.restrictScalars` as a `LinearMap`. -/
+def restrictScalarsₗ : (M →ₗ[S] N) →ₗ[R₁] M →ₗ[R] N where
```
Comment (erdOne) [inline Mathlib/Algebra/Module/LinearMap/Defs.lean:1071]:
```suggestion
@[simps apply]
def restrictScalarsₗ : (M →ₗ[S] N) →ₗ[R₁] M →ₗ[R] N where
```

Comment (erdOne) [reply Mathlib/Algebra/Module/LinearMap/Defs.lean:1071]: This generates a simp lemma `restrictScalarsₗ R l = l.restrictScalars R`

**Comment by erdOne** [comment]
(I've edited the PR message according to the tag you added)

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**